### PR TITLE
Fix support for camera with single channel floating point image format

### PR DESCRIPTION
**Pull request by iche033**
There were places in the Camera and Image class that didn't have proper check for floating point format which could cause bad data to be generated

Signed-off-by: Ian Chen <ichen@osrfoundation.org>

### CODE REVIEW

**Comment by scpeters** [comment]
I just did a quick test and `FLOAT32` images appear correctly in the gzclient topic viewer, but `FLOAT16` images still don't look right

**Comment by scpeters** [comment]
> I just did a quick test and `FLOAT32` images appear correctly in the gzclient topic viewer, but `FLOAT16` images still don't look right

they seem to be interpreted as depth images

* https://github.com/osrf/gazebo/blob/gazebo11/gazebo/gui/viewers/ImageFrame.cc#L71-L92

**Comment by scpeters** [comment]
> > I just did a quick test and `FLOAT32` images appear correctly in the gzclient topic viewer, but `FLOAT16` images still don't look right
> 
> they seem to be interpreted as depth images
> 
> * https://github.com/osrf/gazebo/blob/gazebo11/gazebo/gui/viewers/ImageFrame.cc#L71-L92


https://user-images.githubusercontent.com/3650755/105383409-f14d6c00-5bc5-11eb-916d-c3c2af2c6134.mp4

**Comment by iche033** [comment]
Updated image viewer to visualize 16 bit float images: b76f0e9

**Review by scpeters — APPROVED** [review]

File: Changelog.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 ## Gazebo 9.16.0 (2020-11-24)
 
+1. Fix support for camera with single channel floating point image format
```
Comment (iche033) [inline Changelog.md:7]: just realized I added the changelog entry in the wrong place. It should be in a new `Gazebo 9.xx.xx` section. cc @scpeters, let's remember to move this entry when making a new gazebo 9 release.

Comment (scpeters) [reply Changelog.md:7]: I noticed it right after I merged and fixed with a direct commit to gazebo9: https://github.com/osrf/gazebo/commit/d94e2e0b6754a929c11dec4dff2131cfd291478e

**Pull request merged**